packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -320,6 +331,33 @@ export class RichTextEditor extends RichText implements ErrorPattern {
         return true;
     }
 
+    /**
+     * Toggle the mention node and focus back to the editor
+     * @internal
+     */
+    public mentionButtonClick(character: string): void {
+        this.tiptapEditor.chain().insertContent(` ${character}`).focus().run();
+    }
+
+    /**
+     * Toggle the mention node and focus back to the editor
+     * @internal
+     */
+    public mentionButtonKeyDown(
+        event: KeyboardEvent,
+        character: string
+    ): boolean {
+        if (this.keyActivatesButton(event)) {
+            this.tiptapEditor
+                .chain()
+                .insertContent(` ${character}`)
```

Comment:
Is this space before the character needed? If so ccan you put a comment explaining?
```
                .chain()
                // Need s space for some reason
                .insertContent(` ${character}`)
```

Reply:
Updated the logic, now it will insert space based on `startOfLine` and whitespace on the previous node.